### PR TITLE
Add conditional extension point for testing sidecar functionality

### PR DESCRIPTION
**Pull request by aocenas**
This is an experiment to enable sidecar for Logs + other app and test it internally.

Ties in with https://github.com/grafana/explore-traces/pull/102

Needs `appSidecar` feature toggle and optionally `timeRangeProvider` for time picker sync.

https://github.com/user-attachments/assets/856501fb-6149-4b56-b0d1-bb5c6748c689

### CODE REVIEW

File: src/Components/ServiceScene/ActionBarScene.tsx [hunk 1]
```diff
@@ -39,6 +41,10 @@ export class ActionBarScene extends SceneObjectBase<ActionBarSceneState> {
       <Box paddingY={0}>
         <div className={styles.actions}>
           <Stack gap={1}>
+            {
+              // @ts-ignore appSidecar not yet in stable runtime
```
Comment (aocenas) [inline src/Components/ServiceScene/ActionBarScene.tsx:45]: Not sure how to deal with runtime updates. Should they be part of this or is it better to wait for a release and do something like this in the meantime?

File: src/Components/ServiceScene/ActionBarScene.tsx [hunk 2]
```diff
@@ -100,3 +106,66 @@ function getStyles(theme: GrafanaTheme2) {
...
+      setFilters(newState.filters);
+    });
+    return () => {
+      sub.unsubscribe();
+    };
+  }, [props.serviceScene]);
+
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const extensions = usePluginLinks({
+    extensionPointId: 'grafana-lokiexplore-app/toolbar',
```
Comment (aocenas) [inline src/Components/ServiceScene/ActionBarScene.tsx:131]: Not sure about the naming. Considering the default text in dropdown states "open related" maybe "intent" should be part of the extension point name.

File: src/Components/ServiceScene/ActionBarScene.tsx [hunk 3]
```diff
@@ -39,6 +41,10 @@ export class ActionBarScene extends SceneObjectBase<ActionBarSceneState> {
       <Box paddingY={0}>
         <div className={styles.actions}>
           <Stack gap={1}>
+            {
+              // @ts-ignore appSidecar not yet in stable runtime
+              config.featureToggles.appSidecar && <ToolbarExtensionsRenderer serviceScene={serviceScene} />
```
Comment (aocenas) [inline src/Components/ServiceScene/ActionBarScene.tsx:46]: Not sure this needs to be behind a featureFlag. The code itself does not have any special tie to sidecar, just the reason for adding it now is to test it.

File: src/Components/ServiceScene/ActionBarScene.tsx (hunk 1 above)
Comment (matyax) [inline src/Components/ServiceScene/ActionBarScene.tsx:45]:
```suggestion
              // @ts-expect-error appSidecar not yet in stable runtime
```

Comment (matyax) [reply src/Components/ServiceScene/ActionBarScene.tsx:45]: If you mean having to add the ts-whatever directive, it's okay. We also do that for other features that haven't been released https://github.com/grafana/explore-logs/blob/main/src/Components/ServiceScene/LogsPanelScene.tsx#L122

**Review by matyax — APPROVED** [review]
Looks good, thank you @aocenas !

File: src/Components/ServiceScene/ActionBarScene.tsx (hunk 1 above)
Comment (aocenas) [reply src/Components/ServiceScene/ActionBarScene.tsx]: Hmm actually I think the `@ts-expect-error` is wrong here (well also IDE says so :D). I assume the thing is the `config.featureToggles.appSidecar` does not error out in a regular sense, the `appSidecar` will just be `undefined`.
Screenshot 2024-10-30 at 16 36 33

Comment (matyax) [reply src/Components/ServiceScene/ActionBarScene.tsx]: When we get that it usually means that the feature flag has been released and is present in the type. In any case, it's pretty minor. Thanks for checking!

**Pull request merged**